File: src/Haskell/X86CodeGen.hs
{-# LANGUAGE MultiParamTypeClasses #-}
module X86CodeGen (X86CodeGen(..)) where

import Backend
import X86Instr
import X86Function
import X86Prg

-----------------------------------------
-- Code generation
-----------------------------------------

data X86CodeGen = X86CodeGen

instance CodeGen X86CodeGen X86Prg X86Function X86Instr where
  codeGen X86CodeGen = error "TODO"
  allRegisters X86CodeGen = error "needed later for register allocation"
  generalPurposeRegisters X86CodeGen = error "needed later for register allocation"

File: src/Rust/x86/x86codegen.rs
use backend::CodeGen;
use ident::Ident;
use tree;
use x86::X86Function;
use x86::X86Instr;
use x86::X86Prg;

#[derive(Copy, Clone)]
pub struct X86CodeGen {}

impl CodeGen<X86Instr, X86Function, X86Prg> for X86CodeGen {
    fn all_registers() -> Vec<Ident> {
        panic!("needed later for register allocation")
    }

    fn general_purpose_registers() -> Vec<Ident> {
        panic!("needed later for register allocation")
    }

    fn code_gen(_prg: tree::Prg) -> X86Prg {
        panic!("instruction selection goes here")
    }
}
